clevertap-core/src/main/java/com/clevertap/android/sdk/response/InAppResponse.java

```diff
@@ -136,6 +135,7 @@ public void processResponse(
             InAppImageRepoImpl assetRepo = new InAppImageRepoImpl(cleanupStrategy, preloadStrategy, inAppAssetStore, legacyInAppStore);
             assetRepo.fetchAllImages(res.getPreloadImages());
             assetRepo.fetchAllGifs(res.getPreloadGifs());
+            // TODO CustomTemplates download all file arguments before presenting replace image fetching will general file handling (including custom template files)
```

Comment:
This is code which deals with inapps as a feature, if we need to do a full download handling maybe we will need to add change in specific feature/core modules

Reply:
I've added this TODO just to mark that the downloading should happen when the inapp is processed/received. Not sure that this would be the exact place, it depends on how file downloading would be implemented.